### ISSUE TITLE
breaking change in 1.6.13: Using `*time.Time` as a field value does not properly create/update/select.

### ISSUE THREAD

**Issue by EfChouTR** [snowflakedb/gosnowflake#677]
### Issue description
When using `*time.Time` as a struct field updates/selects no longer updates/returns values to/from the DB. We had to revert to 1.6.12. PR #642 seems to have broken the behavior.

### Example code
```go
type someType struct {
  CreatedAt *time.Time
  ID               string
}
```

define a struct like above, do update/create/select. CreatedAt field is not propagated to the DB. No errors.

### Error log
N/A

### Configuration
*1.6.13*

*Go version:* 1.19.2

*Server version:* 6.34.0

*Client OS:* AmazonLinux2 (CentOS)

**Comment by varmakarthik12** [comment]
Particularly since this fails silently. It took us a while to corner this. Can someone confirm if this is an intentional or a possible bug?

**Comment by EfChouTR** [comment]
This is fixed in #761
